### PR TITLE
[Merged by Bors] - feat: introduce ENat.card, and use it in place of PartENat.card

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/SetTheory/Cardinal/Finite.lean [hunk 1]
```diff
@@ -14,8 +14,10 @@ import Mathlib.SetTheory.Cardinal.PartENat
 
 * `Nat.card α` is the cardinality of `α` as a natural number.
   If `α` is infinite, `Nat.card α = 0`.
+* `ENat.card α` is the cardinality of `α` as an  extended natural number.
+  If `α` is infinite, `ENat.card α = 0`.
```
Comment (ocfnash) [inline Mathlib/SetTheory/Cardinal/Finite.lean:18]:
```suggestion
  If `α` is infinite, `ENat.card α = ⊤`.
```

File: Mathlib/SetTheory/Cardinal/Finite.lean [hunk 2]
```diff
@@ -242,85 +244,176 @@ protected alias ⟨_, Nonempty.natCard_pos⟩ := natCard_pos
...
+@[simp]
+theorem card_eq_top_of_infinite [Infinite α] : card α = ⊤ := by
+  simp [card]
+
+@[simp]
+theorem card_sum (α β : Type*) :
+    card (α ⊕ β) = card α + card β := by
+  simp only [card, mk_sum, map_add, toENat_lift]
+
+theorem card_congr {α : Type*} {β : Type*} (f : α ≃ β) : card α = card β :=
```
Comment (ocfnash) [inline Mathlib/SetTheory/Cardinal/Finite.lean:268]: IIUC each of `α` and `β` still get their own universe even if we write:
```suggestion
theorem card_congr {α β : Type*} (f : α ≃ β) : card α = card β :=
```

File: Mathlib/SetTheory/Cardinal/Finite.lean [hunk 3]
```diff
@@ -242,85 +244,176 @@ protected alias ⟨_, Nonempty.natCard_pos⟩ := natCard_pos
...
+  Cardinal.toENat_congr f
+
+@[simp] lemma card_ulift (α : Type*) : card (ULift α) = card α := card_congr Equiv.ulift
+
+@[simp] lemma card_plift (α : Type*) : card (PLift α) = card α := card_congr Equiv.plift
+
+theorem card_image_of_injOn {α : Type u} {β : Type v} {f : α → β} {s : Set α} (h : Set.InjOn f s) :
```
Comment (ocfnash) [inline Mathlib/SetTheory/Cardinal/Finite.lean:275]:
```suggestion
theorem card_image_of_injOn {α β : Type*} {f : α → β} {s : Set α} (h : Set.InjOn f s) :
```

File: Mathlib/SetTheory/Cardinal/Finite.lean [hunk 4]
```diff
@@ -242,85 +244,176 @@ protected alias ⟨_, Nonempty.natCard_pos⟩ := natCard_pos
...
+    card (f '' s) = card s :=
+  card_congr (Equiv.Set.imageOfInjOn f s h).symm
+
+theorem card_image_of_injective {α : Type u} {β : Type v} (f : α → β) (s : Set α)
```
Comment (ocfnash) [inline Mathlib/SetTheory/Cardinal/Finite.lean:279]:
```suggestion
theorem card_image_of_injective {α β : Type*} (f : α → β) (s : Set α)
```

File: Mathlib/SetTheory/Cardinal/Finite.lean [hunk 5]
```diff
@@ -242,85 +244,176 @@ protected alias ⟨_, Nonempty.natCard_pos⟩ := natCard_pos
...
 namespace PartENat
 
 /-- `PartENat.card α` is the cardinality of `α` as an extended natural number.
   If `α` is infinite, `PartENat.card α = ⊤`. -/
+@[deprecated ENat.card (since := "2024-11-30")]
 def card (α : Type*) : PartENat :=
   toPartENat (mk α)
 
-@[simp]
+-- The remainder of this section is about the depreacted `PartENat.card`.
```
Comment (ocfnash) [inline Mathlib/SetTheory/Cardinal/Finite.lean:334]:
```suggestion
-- The remainder of this section is about the deprecated `PartENat.card`.
```

**Review by ocfnash — APPROVED** [review]
Thanks so much for doing this!

bors d+

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**